packages/react-scripts/scripts/utils/createJestConfig.js

```diff
@@ -46,5 +47,49 @@ module.exports = (resolve, rootDir, isEjecting) => {
   if (rootDir) {
     config.rootDir = rootDir;
   }
+  const overrides = Object.assign({}, require(paths.appPackageJson).jest);
+  if (overrides) {
+    if (overrides.collectCoverageFrom) {
+      config.collectCoverageFrom = overrides.collectCoverageFrom;
+      delete overrides.collectCoverageFrom;
+    }
+    if (overrides.coverageReporters) {
+      config.coverageReporters = overrides.coverageReporters;
+      delete overrides.coverageReporters;
+    }
+    if (overrides.coverageThreshold) {
+      config.coverageThreshold = overrides.coverageThreshold;
+      delete overrides.coverageThreshold;
+    }
+    if (overrides.snapshotSerializers) {
+      config.snapshotSerializers = overrides.snapshotSerializers;
+      delete overrides.snapshotSerializers;
+    }
+    const unsupportedKeys = Object.keys(overrides);
+    if (unsupportedKeys.length) {
+      console.error(
+        chalk.red(
+          'By default, Create React App only supports overriding the following Jest options: ' +
+            chalk.bold('collectCoverageFrom') +
+            ', ' +
+            chalk.bold('coverageReporters') +
+            ', ' +
+            chalk.bold('coverageThreshold') +
+            ', and ' +
+            chalk.bold('snapshotSerializers') +
+            '.' +
+            '\n\nThe following options in your package.json Jest configuration are not currently supported by Create React App:\n\n' +
+            unsupportedKeys
+              .map(key => chalk.bold('  \u2022 ' + key))
+              .join('\n') +
+            '\n\nIf you wish to override other options, you need to eject from the default setup. ' +
```

Comment:
Maybe suggest opening an issue if there's a compelling enough reason to allow overriding?